tests/test_mlkit.py

```diff
@@ -351,9 +481,180 @@ def test_model_source_validation_errors(self, model_source):
+        model.wait_for_unlocked()
+
+    def test_wait_for_unlocked(self):
+        recorder = instrument_mlkit_service(status=200,
+                                            operations=True,
+                                            payload=OPERATION_DONE_PUBLISHED_RESPONSE)
+        model = mlkit.Model.from_dict(LOCKED_MODEL_JSON_1)
+        model.wait_for_unlocked()
+        assert model == FULL_MODEL_PUBLISHED
+        assert len(recorder) == 1
+        assert recorder[0].method == 'GET'
+        assert recorder[0].url == TestModel._op_url(PROJECT_ID, MODEL_ID_1)
+
+    def test_wait_for_unlocked_timeout(self):
+        instrument_mlkit_service(status=200, operations=True, payload=OPERATION_NOT_DONE_RESPONSE)
+        mlkit._MLKitService.POLL_BASE_WAIT_TIME_SECONDS = 5  # longer for timeout
+        model = mlkit.Model.from_dict(LOCKED_MODEL_JSON_1)
+        with pytest.raises(Exception) as excinfo:
+            model.wait_for_unlocked(max_time_seconds=3)
+        check_error(excinfo, exceptions.DeadlineExceededError, 'Polling max time exceeded.')
+
+
+class TestCreateModel(object):
+    """Tests mlkit.create_model."""
+    @classmethod
+    def setup_class(cls):
+        cred = testutils.MockCredential()
+        firebase_admin.initialize_app(cred, {'projectId': PROJECT_ID})
+        mlkit._MLKitService.POLL_BASE_WAIT_TIME_SECONDS = 0.1  # shorter for test
+
+    @classmethod
+    def teardown_class(cls):
+        testutils.cleanup_apps()
+
+    @staticmethod
+    def _url(project_id):
+        return BASE_URL + 'projects/{0}/models'.format(project_id)
+
+    @staticmethod
+    def _op_url(project_id, model_id):
+        return BASE_URL + \
+            'operations/project/{0}/model/{1}/operation/123'.format(project_id, model_id)
+
+    @staticmethod
+    def _get_url(project_id, model_id):
+        return BASE_URL + 'projects/{0}/models/{1}'.format(project_id, model_id)
+
+    def test_immediate_done(self):
+        instrument_mlkit_service(status=200, payload=OPERATION_DONE_RESPONSE)
+        model = mlkit.create_model(MODEL_1)
+        assert model == CREATED_MODEL_1
+
+    def test_with_get_operation(self):
+        create_recorder = instrument_mlkit_service(
+            status=200, payload=OPERATION_NOT_DONE_RESPONSE)
+        operation_recorder = instrument_mlkit_service(
+            status=200, operations=True, payload=OPERATION_DONE_RESPONSE)
+        model = mlkit.create_model(MODEL_1)
+        assert model == CREATED_MODEL_1
+        assert len(create_recorder) == 1
+        assert create_recorder[0].method == 'POST'
+        assert create_recorder[0].url == TestCreateModel._url(PROJECT_ID)
+        assert len(operation_recorder) == 1
+        assert operation_recorder[0].method == 'GET'
+        assert operation_recorder[0].url == TestCreateModel._op_url(PROJECT_ID, MODEL_ID_1)
+
+    def test_with_get_returns_locked(self):
+        recorder = instrument_mlkit_service_multi(
```

Comment:
I think `create_model()` should only ever make 1 RPC. WDYT?

Reply:
I'm not sure how that would work. It would have to be a CreateModel rpc which returns an operation. But create_model always returns a (possibly locked) model... The operation does not have model information if it is still running... 

I think it should do exactly what it does now: 
call the CreateModel api, and up to 1 GetOperation poll because that will give us the "low hanging fruit" for models to not be locked. If the GetOperation still results in a not-done operation, then we need to GetModel to get the model information (not available in the Operations returned) so we can return the locked model.